### ISSUE TITLE
Flag should exist to only test staged files in Git repo

### ISSUE THREAD

**Issue by oschwald** [houseabsolute/perl-code-tidyall#38]
`-g` seems to test everything that has changed.

**Comment by oalders** [comment]
I run into this fairly often when I have a large number of changes, but stage some lines individually.  My commit on a small chunk of code fails because of some unstaged code that doesn't pass tidying/linting.  It would be nice to have a `--staged` flag or something similar to go along with `-g`.  `git cat-file -p :path/to/file` seems to be the syntax to view just the staged file.

Mainly I'd be interested in this as default behaviour when running a git `pre-commit` hook.

**Comment by autarch** [comment]
@oalders I agree that making this behavior the default for the commit hook code makes sense.

Going further, I think it's a lot easier to implement checking only the staged files or portions of files for the `--check-only` mode. While I think in theory we could tidy just the staged parts of a file that's been partially staged, I'm not sure it's worth the effort.

FWIW, I've been vaguely thinking of either a major revision of tidyall or a new successor tool (possibly in Go?). If it were designed from the ground up to deal with this sort of thing (and other issues like Go linters operating on a dir rather than a file) then maybe it'd be more tractable.

**Comment by oalders** [comment]
@autarch that all makes sense.  Just tidying the staged parts might be handy but probably more effort than it's worth.  A Go rewrite might be fun. :)

**Comment by autarch** [comment]
So I did start on a replacement for tidyall. See https://github.com/houseabsolute/precious for my work in progress. Mostly I've been trying to figure out configuration, since that's about 75% of the work for this sort of tool!
